### ISSUE TITLE
Multi selectors

### ISSUE THREAD

**Issue by blond** [bem-incubator/bem-deps#52]
There are cases when several entities depend on other entity. 

The dependency should be added only when both entities are in declaration.

**Example**

```js
var bemDeps = require('@bem/deps');

var relations = {
    entity: { block: 'A' },
    with: [
        { entity: { block: 'B' } }
    ],
    dependOn: [
        { entity: { block: 'C' } }
    ]
};

bemDeps.resolve([{ block: 'A' }, { block: 'B' }], relations);

// [
// { block: 'A' },
// { block: 'B' },
// { block: 'C' }
// ]

bemDeps.resolve([{ block: 'A' }], relations);

// [
// { block: 'A' }
// ]
```

**Real examples**:

1. `{ block: 'checkbox', modName: 'theme', modVal: 'islands'` with `{ block: 'checkbox', modName: 'type', modVal: 'button' }` depend on `{ block: 'button', modName: 'theme', modVal: 'islands' }`.

  `islands` theme of `button` need add only if `checkbox` has `type_button` modifier with `islands` theme.

  Code in `bem-components` — https://github.com/bem/bem-components/blob/5406834d2f35911574ec49f974c0788c800d7835/design/common.blocks/checkbox/_theme/checkbox_theme_islands.deps.js

2. `{ block: 'dropdown', modName: 'theme', modVal: 'islands'` with `{ block: 'dropdown', modName: 'switcher', modVal: 'link' }` depend on `{ block: 'link', modName: 'theme', modVal: 'islands' }`.

  `islands` theme of `link` need add only if `dropdown` has `switcher_link` modifier with `islands` theme.

  Code in `bem-components` — https://github.com/bem/bem-components/blob/5406834d2f35911574ec49f974c0788c800d7835/design/common.blocks/dropdown/_theme/dropdown_theme_islands.deps.js

**Comment by zxqfox** [comment]
```
var relations = {
    entity: { block: 'A' },
    onlyWith: [
        { entity: { block: 'B' } }
    ],
    dependOn: [
        { entity: { block: 'C' } }
    ]
};
```

Т.е. это надо будет дважды прописывать? И для блока A, и для B?

**Comment by zxqfox** [comment]
Real examples could be easily a bad architecture.

```
    onlyWith: [
        { entity: { block: 'B' } }
    ],
```
You sure that we need `entity` here?

Do you specified RelationNode types (or something like that) already?

**Comment by blond** [comment]
> Т.е. это надо будет дважды прописывать? И для блока A, и для B?

Нет. Если в декларации есть A и B то работает, если чего-то нет, то не работает. В какой последовательности записали — неважно.

Смущает синтаксис? Есть идеи как лучше?

**Comment by blond** [comment]
> You sure that we need entity here?

Yes. This is consistent with other cases. And we can specify the technology.

```js
onlyWith: [
    {
        entity: { block: 'B' },
        tech: 'js'
    }
]
```

> Do you specified RelationNode types (or something like that) already?

What do you mean by RelationNode types?

**Comment by zxqfox** [comment]
> Смущает синтаксис? Есть идеи как лучше?
Смущает то, что надо будет помнить про вхождение блока B в зависимостях блока А
Если удаляем блок В, то надо будет лезть во все блоки и чистить зависимости.

**Comment by zxqfox** [comment]
> What do you mean by RelationNode types?

I mean we need specification for node types.

**Comment by blond** [comment]
> Смущает то, что надо будет помнить про вхождение блока B в зависимостях блока А

Так это жизненная необходимость :) Без такого рода зависимостей не избежать избыточного кода.

Такие зависимости можно писать не в файлах блока, а каких-нибудь файлах уровня или проекта. Но это уже дело вкуса.

**Comment by blond** [comment]
> I mean we need specification for node types.

Relations is not graph, although very similar. Therefore, there are no nodes and their types.

Or do you mean that we should describe the relations format?

**Comment by zxqfox** [comment]
@blond The question of missing terminology. What is relation? Where are dependency-edges?
